README.md

```diff
@@ -299,6 +299,48 @@ async function main() {
 main().catch((error) => console.error(error))
 ```
 
+### File Upload
+
+#### Browser
+
+```js
+import { request } from 'graphql-request'
+
+const UploadUserAvatar = gql`
+  mutation uploadUserAvatar($userId: Int!, $file: Upload!) {
+    updateUser(id: $userId, input: { avatar: $file })
+  }
+`
+
+request('/api/graphql', UploadUserAvatar, {
+  userId: 1,
+  file: document.querySelector('input#avatar').files[0],
+})
+```
+
+#### NodeJS
+
+```js
+import 'cross-fetch/polyfill' // fetch polyfill
+
+import { createReadStream } from 'fs'
+import FormData from 'form-data'
+import { request } from 'graphql-request'
+
+global.FormData = FormData
```

Comment:
I think this should be polyfilled automatically. If user doesn't want that, they should be able to pass option to disable it.

I think setting of type ` { polyfill?: { formData?: boolean } }` would work well.

Finally, we should document/mentioned this.

Reply:
I'm not sure about this... 

1) More dependencies. Everyone will have `form-data` installed but not everyone is using file upload and not everyone is using Node.js
2) Third option for `request`. I may forget about it in some requests since it's optional and `form-data` will be polyfilled automatically which may be unexpected
3) Global space pollution inside a library seems like an unexpected side effect

We can just import `form-data` and use it inside library without global space pollution if `global.FormData` is undefined. It'll solve 2) and 3)

Reply:
Yeah that's better, and what we do with fetch so more consistent approach.

Reply:
Ok, done